admin/class-admin-init.php

```diff
@@ -430,11 +430,21 @@ private function load_admin_page_class() {
 			// Only register the yoast i18n when the page is a Yoast SEO page.
 			if ( WPSEO_Utils::is_yoast_seo_free_page( filter_input( INPUT_GET, 'page' ) ) ) {
 				$this->register_i18n_promo_class();
+				$this->register_premium_upsell_admin_block();
 			}
 		}
 	}
 
 	/**
+	 * Registers the Premium Upsell Admin Block.
+	 *
+	 * @return void
+	 */
+	private function register_premium_upsell_admin_block() {
+		new Premium_Upsell_Admin_Block( 'wpseo_admin_promo_footer' );
```

Comment:
It would be nice to implement `WPSEO_WordPress_Integration` and call the register_hooks separately.